openrr-teleop/src/control_nodes_config.rs

```diff
@@ -115,6 +150,228 @@ impl ControlNodesConfig {
+            step_duration_secs: default_step_duration_secs(),
+        };
+        let _joy_tconfig = JoyJointTeleopConfig {
+            client_name: String::from("b"),
+            config,
+        };
+
+        // IkNodeTeleopConfig
+        let config = IkNodeConfig {
+            mode: String::from("c"),
+            move_step_angular: default_move_step_angular(),
+            move_step_linear: default_move_step_linear(),
+            step_duration_secs: default_step_duration_secs(),
+        };
+
+        let _iknode_tconfig = IkNodeTeleopConfig {
+            solver_name: String::from("d"),
+            joint_trajectory_client_name: String::from("f"),
+            config,
+        };
+    }
+
+    #[test]
+    fn test_control_node_config_create_error() {
+        let jj_config = JoyJointTeleopConfig {
+            client_name: String::from("jj config"),
+            config: JoyJointTeleopNodeConfig {
+                mode: String::from("jjt config"),
+                joint_step: 1.0_f64,
+                step_duration_secs: default_step_duration_secs(),
+            },
+        };
+        let ik_config = IkNodeTeleopConfig {
+            solver_name: String::from("ik config solver"),
+            joint_trajectory_client_name: String::from("ik trajectory"),
+            config: IkNodeConfig {
+                mode: String::from("ik node config"),
+                move_step_angular: default_move_step_angular(),
+                move_step_linear: default_move_step_linear(),
+                step_duration_secs: default_step_duration_secs(),
+            },
+        };
+        let rc_config = RobotCommandConfig {
+            name: String::from("rc config"),
+            file_path: String::from("sample path"),
+        };
+        let ctrl_node_config = ControlNodesConfig {
+            move_base_mode: Some(String::from("a")),
+            joy_joint_teleop_configs: vec![jj_config],
+            ik_node_teleop_configs: vec![ik_config],
+            joints_pose_sender_config: None,
+            command_configs: vec![rc_config],
+        };
+
+        // Check create control node(Error:HashMap length)
+        let speaker = Arc::new(DummySpeaker::default());
+        let robot_client = ArcRobotClient::new(
+            OpenrrClientsConfig::default(),
+            HashMap::new(),
+            HashMap::new(),
+            None,
+            None,
+            None,
+        );
+        assert!(robot_client.is_ok());
+        let robot_client = robot_client.unwrap();
+        let nodes = ctrl_node_config.create_control_nodes(
+            None,
+            Arc::new(robot_client),
+            speaker.clone(),
+            &HashMap::new(),
+            &HashMap::new(),
+            Some(Arc::new(DummyMoveBase::default())),
+            Vec::new(),
+        );
+        assert!(nodes.is_err());
+        let err = nodes.err().unwrap();
+        println!("{}", err);
+
+        // Check create control node(Error:name mismatch)
```

Comment:
Explain in more detail.
I want to know which parameter caused the error. (The same applies above.)